framework/include/Logger.h

```diff
@@ -0,0 +1,41 @@
+#ifndef _LOGGER_H
+#define _LOGGER_H
+
+#include <fstream>
+#include <map>
+#include <memory>
+#include <mutex>
+
+namespace Logging 
+{
+
+static const std::string PERFORMANCE_LOG = "Performance_Log";
+static const std::string SIMULATION_LOG = "Simulation_Log";
+static const std::string TEST_LOG = "Test_Log";
+
+// Definition of the Logger class
+class Logger 
+{
+public:
+    Logger(Logger const&) = delete;
+    void operator=(Logger const&) = delete;
+
+    // Logger Instance
+    static Logger& getInstance() 
```

Comment:
One option instead of using a singleton is to just declare a static instance using static const under the class declaration. That removes the step of calling getInstance. It isn't a big deal though, just a thought.